### PR TITLE
add xdg-desktop-portal-kde-5.13.3

### PR DESCRIPTION
**Pull request by seiflek**
This provides the xdg-desktop-portal-kde that i wrote couple days ago, i've tested it for quite a bit, but still needs some more testing and some feedback.

### CODE REVIEW

File: sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit cmake-utils
+
+SRC_URI="mirror://kde/stable/plasma/${PV}/${P}.tar.xz"
+DESCRIPTION="A backend implementation for xdg-desktop-portal using Qt/KF5"
+HOMEPAGE="https://www.kde.org/workspaces/plasmadesktop/"
```
Comment (fosero) [inline sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild]: I would prefer if this points to the actual git repo.

File: sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+DEPEND="
+	>=kde-frameworks/extra-cmake-modules-5.42.0
+	>=kde-frameworks/ki18n-5.42.0
+	>=kde-frameworks/knotifications-5.42.0
+	>=kde-frameworks/kcoreaddons-5.42.0
+	>=kde-frameworks/kwidgetsaddons-5.42.0
+	wayland? (
+		>=kde-frameworks/kwayland-5.42.0
+		)
```
Comment (fosero) [inline sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild]: I don't see wayland as optional in the cmake file?

Also, make this a single line:

```sh
wayland? ( >=kde-frameworks/kwayland-5.42.0 )
```

File: sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	>=dev-qt/qtcore-5.9.0
+	>=dev-qt/qtgui-5.9.0
+	>=dev-qt/qtwidgets-5.9.0
+	>=dev-qt/qtdbus-5.9.0
+"
```
Comment (fosero) [inline sys-apps/xdg-desktop-portal-kde/xdg-desktop-portal-kde-5.13.3.ebuild]: I seem to miss some deps mentioned in cmake file (eg. concurrent), but I am not up-to-date on qt/kde.

**Review by fosero — CHANGES_REQUESTED** [review]
I don't mind adding this, but I am no KDE user at all. So I can't test.

It would need to be updated by you (or anyone willing) to keep it in the tree over time.

**Comment by seiflek** [comment]
Making kde-frameworks/kwayland and dev-qt/qtconcurrent a strict deps for it seems fine, although i think the package should of existed with the kde stack in Gentoo upstream tree and packaged in a more proper manner using `inherit kde5` with the current deps in order to integrate tightly with the rest of the stack.

`DEPEND="`
`    $(add_frameworks_dep extra-cmake-modules)`
`    $(add_frameworks_dep ki18n)`
`    $(add_frameworks_dep knotifications)`
`    $(add_frameworks_dep kcoreaddons)`
`    $(add_frameworks_dep kwidgetsaddons)`
`    $(add_frameworks_dep kwayland)`
`    $(add_qt_dep qtcore)`
`    $(add_qt_dep qtgui)`
`    $(add_qt_dep qtwidgets)`
`    $(add_qt_dep qtconcurrent)`
`    $(add_qt_dep qtdbus)`
`    $(add_qt_dep qtprintsupport)`
`"`

I did package it for the sake of filling the gap, cause i saw a lot of people requesting it. And for upgrading and testing it, i'll take on it until the kde-team in Gentoo decide to implement it upstream, which i think will be the case soon since it's officially part of the kde familly.

**Comment by seiflek** [comment]
I also think sys-app/ is not the right place for it, it should exist in kde-plasma/

**Comment by fosero** [comment]
Did you mean to close this PR?

**Comment by seiflek** [comment]
i re-wrote the ebuild to use `inherit kde5` to better integrate with KF version provided in Gentoo upstream tree. i'll re-submit it in another pull request.

**Pull request closed without merging**